src/main/kotlin/io/github/gradlenexus/publishplugin/DefaultNexusRepositoryContainer.kt

```diff
@@ -21,22 +21,15 @@ import org.gradle.api.Action
 import org.gradle.api.NamedDomainObjectContainer
 import org.gradle.api.internal.NamedDomainObjectContainerConfigureDelegate
 import org.gradle.util.ConfigureUtil
-import java.net.URI
 
 @Suppress("UnstableApiUsage")
 internal class DefaultNexusRepositoryContainer(delegate: NamedDomainObjectContainer<NexusRepository>) : NexusRepositoryContainer, NamedDomainObjectContainer<NexusRepository> by delegate {
 
     override fun sonatype(): NexusRepository = sonatype {}
 
-    override fun sonatype(closure: Closure<*>): NexusRepository {
-        return sonatype(ConfigureUtil.configureUsing(closure))
-    }
+    override fun sonatype(closure: Closure<*>): NexusRepository = sonatype(ConfigureUtil.configureUsing(closure))
```

Comment:
Closure overloads are not needed, and they can be removed

Reply:
Why are these not needed any more?

Anyhow I'd prefer to keep this out of the scope of this PR: It would change the `NexusRepositoryContainer` interface as well and should be done with caution.

Reply:
Methods with `Action<...>` work for ages for both Groovy DSL and Kotlin DSL, so there's no need to duplicate methods, and if only `Action<...>` methods are present, then there's less likely there will be something wrong with ConfigureUtil.

---

> Anyhow I'd prefer to keep this out of the scope of this PR

I see you touch Closure-related line, so I would suggest just removing it instead of trying to maintain it.

Reply:
I acutally prefer a `Closure` over an `Action` when using the _Groovy_ DSL 😅

Reply:
@tynn ,
1) Gradle makes sure Action works for Groovy just fine
2) `Closure<*>` provides **no** clue on the receiver type. In other words, `Closure<*>` is like Java 1.4 with `Object`, `Vector`, `Hashtable` , and so on.
3) Lots of Gradle developers recommend stopping using `Closure` in plugin apis. For instance: https://twitter.com/CedricChampeau/status/1349669642646806531

Reply:
See https://github.com/gradle/gradle/blob/732148799d2e7b0efe07b2649d5b80b4b3ab7c3f/subprojects/model-core/src/main/java/org/gradle/util/ConfigureUtil.java#L37

Reply:
I filed https://github.com/gradle/gradle/issues/18973 so Gradle issues a warning in case developers use Closure or T.() for the API.

Reply:
This is required due to https://github.com/gradle/gradle/issues/9987

Reply:
It looks like `publish-plugin` uses `newInstance` explicitly: https://github.com/gradle-nexus/publish-plugin/blob/0aea95597e2eeaf9275a286254894bad29b3d1e1/src/main/kotlin/io/github/gradlenexus/publishplugin/NexusPublishExtension.kt#L53-L55, so `#9987` should not be important